svc/GridExportService.js

```diff
@@ -131,6 +142,58 @@ export class GridExportService extends HoistService {
         }
     }
 
+    /**
+     * Get the exportable value for a given cell.
+     *
+     * This method is used internally by this service, but also made available
+     * publicly for use by grid clipboard functionality.
+     *
+     * @param {Object} c
+     * @param {GridModel} c.gridModel
+     * @param {Record} c.record
+     * @param {Column} c.column
+     * @param {Object} c.node - ag-Grid row
+     * @param {boolean} [c.forServer] - for posting to server, default false.
+     * @return {String} - value suitable for export to excel, csv, or clipboard.
+     */
+    getExportableValueForCell({gridModel, record, column, node, forServer = false}) {
+        const {field, exportValue, getValueFn} = column,
+            aggData = gridModel.treeMode && !isEmpty(record.children) ? node.aggData : null;
+
+        let value = getValueFn({record, field, column, gridModel});
+        // Modify value using exportValue
+        if (isString(exportValue) && record.data[exportValue] !== null) {
+            // If exportValue points to a different field
+            value = record.data[exportValue];
+        } else if (isFunction(exportValue)) {
+            // If export value is a function that transforms the value
+            value = exportValue(value, {record, column, gridModel});
+        } else if (aggData && !isNil(aggData[field])) {
+            // If we found aggregate data calculated by agGrid
+            value = aggData[field];
+        }
+
+        if (isNil(value)) return null;
+
+        // Get cell-level format if function form provided
+        let {exportFormat} = column,
+            cellHasExportFormat = isFunction(exportFormat);
+
+        if (cellHasExportFormat) {
+            exportFormat = exportFormat(value, {record, column, gridModel});
+        }
+
+        // Enforce date formats expected by server
+        if (exportFormat === ExportFormat.DATE_FMT) value = fmtDate(value);
```

Comment:
Should these only be applied if `forServer === true`? I'm thinking no - I would want a formatted value in my clipboard, not the `Date.toString()` value, so maybe the comment above is just confusing? This is also inconsistent in that we apply date formatting to date fields when copying, but not to other field types - though I think it does make sense.